.github/workflows/checks.yml

```diff
@@ -37,6 +37,15 @@ jobs:
       - name: Compare generated Dashboard vs. Backed up Dashboard
         run: |
           diff -sNrdu /tmp/dashboard.js web/gui/dashboard.js
+  clang-checks:
+    name: Clang
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        run: |
+          docker build -f .github/dockerfiles/Dockerfile.clang .
```

Comment:
This should probably be a separate job under the builder checks, ideally with the same set of test cases that that uses.

Reply:
You want to run this check against very OS/Distro/Version we currently do for the Build / Install checks? That seems overkill IHMO we just want ot ensure we correctly build with Clang. In future we should also take that binary and run a suite of integration tests against it.

Reply:
No, I want to test the same set of cases that we verify on each of the distros we do build tests on.  Effectively, a manual build (`./configure && make`), a build with the cloud disabled, and a build with it force enabled (and possibly one with the cloud force enabled but no JSON-C on the local system).

It also makes more sense to have it in with the builder checks than here, because it is a build test.

Reply:
No. I won't do that here. Not now. We're going to have to rethink our CI as what you're asking for (_rightfully so and we talked about this in Greece too_) is one of our goals. Please remind me at some point to write/draw up a Roadmap for CI/CD/Packaging for Core-SRE I need to so we're on the same page with where we're going and so others can also poke their heads in and have a look at our "big goals".

Long term we're going to have to come up with a way to do CI/CD in more complex ways than we do now with a whole barrage of setups and configurations with e2e tests and stress tests hammered at them. I expect these tests to take significant time to complete however and there will be many of them. I don't want to a) Block PR(s) unnecessary or b) Clutter up the already cluttered checks we have today for no good reason.

One idea I have in mind is to kick off an external job that does all of this in something like [Argo](https://argoproj.github.io/argo-cd/)[Gaia](https://www.gaia-pipeline.io/).